spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/KinesisMessageChannelBinder.java

```diff
@@ -286,6 +352,37 @@ protected MessageProducer createConsumerEndpoint(ConsumerDestination destination
 		return adapter;
 	}
 
+	private MessageProducer createKclConsumerEndpoint(ConsumerDestination destination, String group,
+			ExtendedConsumerProperties<KinesisConsumerProperties> properties) {
+		KinesisConsumerProperties kinesisConsumerProperties = properties.getExtension();
+
+		String shardIteratorType = kinesisConsumerProperties.getShardIteratorType();
+
+		KclMessageDrivenChannelAdapter adapter = new KclMessageDrivenChannelAdapter(
+			destination.getName(), this.amazonKinesis, this.cloudWatchClient, this.dynamoDBClient, awsCredentialsProvider);
+		adapter.setExecutor(kclTaskExecutor);
+
+		boolean anonymous = !StringUtils.hasText(group);
+		String consumerGroup = anonymous ? "anonymous." + UUID.randomUUID().toString() : group;
+		adapter.setConsumerGroup(consumerGroup);
+
+		if (StringUtils.hasText(shardIteratorType)) {
+			adapter.setStreamInitialSequence(InitialPositionInStream.valueOf(shardIteratorType));
+		}
+
+		adapter.setIdleBetweenPolls(kinesisConsumerProperties.getIdleBetweenPolls());
+		adapter.setConsumerBackoff(kinesisConsumerProperties.getConsumerBackoff());
+		if (this.configurationProperties.getCheckpoint().getInterval() != null) {
+			adapter.setCheckpointsInterval(this.configurationProperties.getCheckpoint().getInterval());
```

Comment:
Please, adapt this `interval` feature in the regular `createKinesisConsumerEndpoint()` as well.